### PR TITLE
Added new bot 

### PR DESCRIPTION
**Pull request by RavidAtias**
* modify_network_security_group_scope_by_port

### CODE REVIEW

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# Example: AUTO: modify_network_security_group_scope_by_port 556 source 10.0.0.0/24,172.16.0.1/32,168.243.22.0/23
+# Limitations: None
+
+from msrestazure.azure_exceptions import CloudError
+from azure.mgmt.network import NetworkManagementClient
+import logging
+
+
+def modify_scope(rule, direction, scope):
+    if len(scope) == 1:
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: len(scope) == 1 should be a variable!

Comment (udidome9) [reply dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: 1 should be a consts

Comment (udidome9) [reply dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: e.g.

MUST_HAPPEN = 1
isItHappen = len(scope) == MUST_HAPPEN
if isItHappen:
pass

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        rule[f'{direction}_address_prefix'] = scope[0]
+        rule[f'{direction}_address_prefixes'] = None
+    else:
+        rule[f'{direction}_address_prefixes'] = scope
+        rule[f'{direction}_address_prefix'] = None
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: rule[f'{direction}_address_prefix'] = None happen on both cases 
why not use it once ?

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def is_port_in_range(port_to_find, ports_list):
+    for port in ports_list:
+        if port == port_to_find:
+            return True
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: new line after each return (all over the file)

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        else:
+            port = port.split('-')
+            if len(port) > 1 and port_to_find> port[0] and port_to_find< port[1]:
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: unpack the port do not use port[0] port[1]

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+def run_action(credentials, rule, entity, params):
+    port, direction, scope = params
+    scope = scope.split(',')
+    logging.info(f'{__file__} - run_action started')
+    subscription_id = entity.get('accountNumber')
+    resource_group_name = entity.get('resourceGroup', {}).get('name')
+    nsg_name = entity.get('name')
+    logging.info(
+        f'{__file__} - subscription_id : {subscription_id} - group_name : {resource_group_name} nsg_name : {nsg_name}')
+    try:
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: new line

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        # Change nsg - NetworkSecurityGroup to dict
+        nsg_dict = nsg.as_dict()
+        security_rules = nsg_dict.get('security_rules')
+        for rule in security_rules:
+            if rule.get('destination_port_range'):
+                if rule.get('destination_port_range') == port:
+                    modify_scope(rule, direction, scope)
+            elif rule.get('destination_port_ranges'):
+                if is_port_in_range(port, rule.get('destination_port_ranges')):
+                    modify_scope(rule, direction, scope)
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: both id else are aiming to activate modify_scope(rule, direction, scope) 
why not use or

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        nsg_dict['security_rules'] = security_rules
+        nsg_after_change = nsg.from_dict(nsg_dict)
+        network_client.network_security_groups.create_or_update(resource_group_name, nsg_name, nsg_after_change)
+        id = entity.get('id')
```
Comment (udidome9) [inline dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: you ran over built in function ```id``` do not ran over safe words 
id should be distinguished as ID better to add a var into it 
entityID = entity.get('id')

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py (hunk 2 above)
Comment (RavidAtias) [reply dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: There is :
'{direction}_address_prefixes'
'{direction}_address_prefix'

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py (hunk 6 above)
Comment (RavidAtias) [reply dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: Also here there is difference between one port : rule.get('destination_port_range') and more then one port  rule.get('destination_port_ranges')

File: dome9CloudBots/bots/modify_network_security_group_scope_by_port.py (hunk 7 above)
Comment (RavidAtias) [reply dome9CloudBots/bots/modify_network_security_group_scope_by_port.py]: I changed var name id to entity_id, i hope you meant that

**Pull request merged**